### ISSUE TITLE
Aliases of precommand with flag that takes an argument

### ISSUE THREAD

**Issue by phy1729** [zsh-users/zsh-syntax-highlighting#540]
```
alias sdu='sudo -u'
sdu phy1729 echo foo
```

Should highlight phy1729 as normal (as it's an argument) and echo as a command. The parser currently treats phy1729 as the command.

**Comment by danielshahaf** [comment]
I'm surprised `phy1729` is taken to be a command word.

**Comment by phy1729** [comment]
The issue is that even once precommands is merged only the first word of an alias is run through the state machine.

**Comment by danielshahaf** [comment]
Duplicate of newer #552.
